### PR TITLE
New persisted arbitrable config

### PR DESCRIPTION
**Pull request by Corantin**
Trigger vercel deployement against

### CODE REVIEW

File: pkg/contracts/src/ProxyOwnableUpgrader.sol [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+contract ProxyOwnableUpgrader is OwnableUpgradeable, UUPSUpgradeable {
+    error CallerNotOwner(address _caller, address _owner);
+
+    function initialize(address initialOwner) public initializer {
+        _transferOwnership(initialOwner);
+    }
+
+    function owner() public view override returns (address) {
+        address currentOwner = OwnableUpgradeable.owner();
```
Comment (kamikazebr) [inline pkg/contracts/src/ProxyOwnableUpgrader.sol:15]: call the function as `owner()` instead, that will do external call unecessary, can be internal call

Comment (Corantin) [reply pkg/contracts/src/ProxyOwnableUpgrader.sol:15]: Im not sure I can since the virtual base is public

File: pkg/contracts/src/ProxyOwner.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract ProxyOwner is OwnableUpgradeable {
+    function initialize(address initialOwner) public initializer {
+        _transferOwnership(initialOwner);
+    }
+}
```
Comment (kamikazebr) [inline pkg/contracts/src/ProxyOwner.sol:10]: that contract can be deleted, right?

File: pkg/contracts/src/ProxyOwnableUpgrader.sol [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import {OwnableUpgradeable} from "openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+
+contract ProxyOwnableUpgrader is OwnableUpgradeable, UUPSUpgradeable {
+    error CallerNotOwner(address _caller, address _owner);
+
+    function initialize(address initialOwner) public initializer {
+        _transferOwnership(initialOwner);
+    }
```
Comment (kamikazebr) [inline pkg/contracts/src/ProxyOwnableUpgrader.sol:12]: must call initialize

File: pkg/contracts/src/RegistryCommunityV0_0.sol [hunk 4]
```diff
@@ -219,11 +211,13 @@ contract RegistryCommunityV0_0 is
         address[] strategies;
     }
 
-    function initialize(RegistryCommunityV0_0.InitializeParams memory params, address _collateralVaultTemplate)
-        public
-        initializer
-    {
-        __Ownable_init();
```
Comment (kamikazebr) [inline pkg/contracts/src/RegistryCommunityV0_0.sol:226]: init normally need be called

File: pkg/contracts/src/RegistryFactoryV0_0.sol [hunk 5]
```diff
@@ -47,17 +47,20 @@ contract RegistryFactoryV0_0 is OwnableUpgradeable, UUPSUpgradeable {
...
 
     function initialize(
+        address _owner,
         address _gardensFeeReceiver,
         address _registryCommunityTemplate,
+        address _strategyTemplate,
         address _collateralVaultTemplate
     ) public virtual initializer {
-        __Ownable_init();
+        ProxyOwnableUpgrader.initialize(_owner);
```
Comment (kamikazebr) [inline pkg/contracts/src/RegistryFactoryV0_0.sol:56]: make sure initialize call ownable_init

File: pkg/contracts/test/ProxyOwnerTest.t.sol [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    address anotherWallet = makeAddr("anotherWallet");
+    address protocolFeeReceiver = makeAddr("multisigReceiver");
+
+    function test_upgradeWithProxyAdmin() public {
+        vm.startPrank(deployerWallet);
+
+        ProxyOwner proxyOwner = ProxyOwner(
+            address(
+                new ERC1967Proxy(
+                    address(new ProxyOwner()), abi.encodeWithSelector(ProxyOwner.initialize.selector, deployerWallet)
```
Comment (kamikazebr) [inline pkg/contracts/test/ProxyOwnerTest.t.sol:27]: ah i see ProxyOwner its the contract that should implement the interface? so we need rename the interface to start with I

File: pkg/contracts/test/ProxyOwnerTest.t.sol [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            abi.encodeWithSelector(RegistryFactoryV0_1.initializeV2.selector, deployerWallet)
+        );
+    }
+
+    // function test_Revert_transferProxyAdminOwnership() public {
+    //     vm.startPrank(deployerWallet);
+
+    //     ProxyOwner proxyOwner = ProxyOwner(
+    //         address(
+    //             new ERC1967Proxy(
```
Comment (kamikazebr) [inline pkg/contracts/test/ProxyOwnerTest.t.sol:78]: missing test that revert and not allow normall address upgrade, that are critical tests to have. @kafann @Corantin

File: pkg/contracts/src/ProxyOwnableUpgrader.sol (hunk 3 above)
Comment (Corantin) [reply pkg/contracts/src/ProxyOwnableUpgrader.sol:12]: initialize is just transfering to msg.sender no ?

File: pkg/contracts/src/RegistryFactoryV0_0.sol (hunk 5 above)
Comment (Corantin) [reply pkg/contracts/src/RegistryFactoryV0_0.sol:56]: Implication of that will be OwnerShipTransfered event will be emitted twice 0x -> sender -> specified owner

File: pkg/contracts/src/RegistryFactoryV0_1.sol [hunk 8]
```diff
@@ -26,7 +24,7 @@ contract RegistryFactoryV0_1 is RegistryFactoryV0_0 {
     /*|                 MODIFIERS                  |*/
     /*|--------------------------------------------|*/
 
-    function initializeV2() public reinitializer(2) {
-        __Ownable_init();
+    function initializeV2(address _owner) public reinitializer(2) {
+        transferOwnership(_owner);
```
[the remark this answers is not in the document]
Comment (Corantin) [reply pkg/contracts/src/RegistryFactoryV0_1.sol:28]: see my previous comment

File: pkg/contracts/src/RegistryFactoryV0_0.sol (hunk 5 above)
Comment (Corantin) [reply pkg/contracts/src/RegistryFactoryV0_0.sol:56]: Oh yeah just realized I can pass this owner directly to the init function

File: pkg/contracts/test/ProxyOwnerTest.t.sol (hunk 6 above)
Comment (Corantin) [reply pkg/contracts/test/ProxyOwnerTest.t.sol:27]: its not an interface its a concrete contract that simply implement OwnableUpgreadable

File: pkg/contracts/src/ProxyOwner.sol (hunk 2 above)
Comment (Corantin) [reply pkg/contracts/src/ProxyOwner.sol:10]: No because OnwableUpgradable is abstract

File: pkg/contracts/src/ProxyOwnableUpgrader.sol (hunk 3 above)
Comment (Corantin) [reply pkg/contracts/src/ProxyOwnableUpgrader.sol:12]: Not needed

File: pkg/contracts/test/ProxyOwnerTest.t.sol (hunk 6 above)
Comment (Corantin) [reply pkg/contracts/test/ProxyOwnerTest.t.sol:27]: ProxyOwner is a concrete class

File: pkg/contracts/src/ProxyOwnableUpgrader.sol (hunk 3 above)
Comment (Corantin) [reply pkg/contracts/src/ProxyOwnableUpgrader.sol:12]: The initializer does the set initialized to true:

File: pkg/contracts/src/RegistryCommunityV0_0.sol (hunk 4 above)
Comment (Corantin) [reply pkg/contracts/src/RegistryCommunityV0_0.sol:226]: https://github.com/1Hive/gardens-v2/pull/361#discussion_r1726171409

File: pkg/contracts/src/RegistryFactoryV0_0.sol (hunk 5 above)
Comment (Corantin) [reply pkg/contracts/src/RegistryFactoryV0_0.sol:56]: https://github.com/1Hive/gardens-v2/pull/361#discussion_r1726171409

File: pkg/contracts/src/RegistryFactoryV0_1.sol (hunk 8 above)
[the remark this answers is not in the document]
Comment (Corantin) [reply pkg/contracts/src/RegistryFactoryV0_1.sol:28]: https://github.com/1Hive/gardens-v2/pull/361#discussion_r1726171409

File: pkg/contracts/test/ProxyOwnerTest.t.sol (hunk 7 above)
Comment (Corantin) [reply pkg/contracts/test/ProxyOwnerTest.t.sol:78]: https://github.com/1Hive/gardens-v2/pull/361#discussion_r1726171409

File: pkg/contracts/src/ProxyOwner.sol (hunk 2 above)
Comment (Corantin) [reply pkg/contracts/src/ProxyOwner.sol:10]: So I need this one to instanciate

**Pull request merged**